flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -385,11 +385,16 @@ public Task(
                 shuffleEnvironment.createShuffleIOOwnerContext(
                         taskNameWithSubtaskAndId, executionId, metrics.getIOMetricGroup());
 
+        boolean overdraftBufferEnabled =
+                !nameOfInvokableClass.equals(
+                        "org.apache.flink.streaming.runtime.tasks.SourceStreamTask");
```

Comment:
I wonder if it would have been better to have:

-  `LocalBufferPool#maxOverdraftBuffersPerGate` mutable (non-final)
- provide setter `LocalBufferPool#setMaxOverdraftBuffers`
- call this setter from within `SourceStreamTask`

?

Reply:
Hi @pnowojski , thanks for your suggestion.

It can be considered, I tried, it has some advantages that no need to modify the Task, only the SourceStreamTask, but it will break more interfaces.

We need call `ResultPartition#setMaxOverdraftBuffers in SourceStreamTask`  and ResultPartition will call `LocalBufferPool#setMaxOverdraftBuffers`.

Call chain:
- `recordWriter.getRecordWriter(n).getTargetPartition().setMaxOverdraftBuffers(x)`
-  `LocalBufferPool#setMaxOverdraftBuffers`

The getTargetPartition need to be changed, and need to add 2 setMaxOverdraftBuffers.

I think both are fine, what do you think?

Reply:
Updated.